engine/src/main/java/com/redhat/cloud/notifications/processors/drawer/DrawerProcessor.java

```diff
@@ -0,0 +1,138 @@
+
+@ApplicationScoped
+public class DrawerProcessor extends SystemEndpointTypeProcessor {
+
+    @Inject
+    TemplateRepository templateRepository;
+
+    @Inject
+    TemplateService templateService;
+
+    @Inject
+    BaseTransformer baseTransformer;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Inject
+    EndpointRepository endpointRepository;
+
+    @Inject
+    DrawerNotificationRepository drawerNotificationRepository;
+
+    @Inject
+    EventRepository eventRepository;
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
+    @Override
+    public void process(Event event, List<Endpoint> endpoints) {
+        if (!featureFlipper.isDrawerEnabled()) {
+            return;
+        }
+        if (endpoints != null && !endpoints.isEmpty()) {
+            Set<User> userList = getRecipientList(event, endpoints, EmailSubscriptionType.DRAWER);
+            if (null != userList && !userList.isEmpty()) {
+                process(event, userList);
+            }
+        }
+
+    }
+
+    private void process(Event event, Set<User> userList) {
+        UUID historyId = UUID.randomUUID();
+        Endpoint endpoint = endpointRepository.getOrCreateDefaultSystemSubscription(event.getAccountId(), event.getOrgId(), EndpointType.DRAWER);
+        Log.infof("Processing drawer notification [orgId=%s, eventId=%s, historyId=%s]",
+            endpoint.getOrgId(), event.getId(), historyId);
+
+        long startTime = System.currentTimeMillis();
+
+        NotificationHistory history = null;
+        try {
+            String userNameListAsStr = userList.stream().map(usr -> usr.getUsername()).collect(Collectors.joining(","));
+            List<DrawerNotification> drawerNotifications = drawerNotificationRepository.create(event, userNameListAsStr);
+
+            // build event thought qute template
+            String renderedData = buildNotificationMessage(event);
+            // store it on event table
+            event.setRenderedDrawerNotification(renderedData);
+            eventRepository.updateDrawerNotification(event);
+            endpoint = endpointRepository.getOrCreateDefaultSystemSubscription(event.getAccountId(), event.getOrgId(), EndpointType.DRAWER);
+
+            // TODO push created drawerNotifications through kafta: RHCLOUD-25999
+
+            history = getHistoryStub(endpoint, event, 0L, historyId);
+            history.setStatus(NotificationStatus.SUCCESS);
+        } catch (Exception e) {
+            history = getHistoryStub(endpoint, event, 0L, historyId);
+            history.setStatus(NotificationStatus.FAILED_INTERNAL);
+            history.setDetails(Map.of("failure", e.getMessage()));
+            Log.infof(e, "Processing drawer notification failed [eventId=%s, historyId=%s]", event.getId(), historyId);
+        } finally {
+            long invocationTime = System.currentTimeMillis() - startTime;
+            history.setInvocationTime(invocationTime);
```

Comment:
I'm not sure the invocation time makes sense for the Drawer since we will only push a message to Kafka which is always instant.

Reply:
You're right, it could be useful even though the name would be a bit misleading since we're not invoking anything 😃 Let's keep it!